src/som/interpreter/Interpreter.java

```diff
@@ -50,176 +52,211 @@ public Interpreter(final Universe universe) {
     this.universe = universe;
   }
 
-  private void doDup(final Frame frame) {
-    // Handle the DUP bytecode
-    frame.push(frame.getStackElement(0));
+  private void doDup(final VirtualFrame frame)
+      throws FrameSlotTypeException {
+    SAbstractObject value = StackUtils.getRelativeStackElement(frame, 0);
+
+    StackUtils.push(frame, value);
   }
 
-  private void doPushLocal(final int bytecodeIndex, final Frame frame, final SMethod method) {
-    // Handle the PUSH LOCAL bytecode
-    frame.push(
-        frame.getLocal(method.getBytecode(bytecodeIndex + 1),
-            method.getBytecode(bytecodeIndex + 2)));
+  private void doPushLocal(final int bytecodeIndex,
+      final VirtualFrame frame,
+      final SMethod method) throws FrameSlotTypeException {
+
+    SAbstractObject value = StackUtils.getLocal(frame,
+        method.getBytecode(bytecodeIndex + 1),
+        method.getBytecode(bytecodeIndex + 2));
+
+    StackUtils.push(frame, value);
+
   }
 
-  private void doPushArgument(final int bytecodeIndex, final Frame frame,
-      final SMethod method) {
-    // Handle the PUSH ARGUMENT bytecode
-    frame.push(
-        frame.getArgument(method.getBytecode(bytecodeIndex + 1),
-            method.getBytecode(bytecodeIndex + 2)));
+  private void doPushArgument(final int bytecodeIndex, final VirtualFrame frame,
+      final SMethod method) throws FrameSlotTypeException {
+    SAbstractObject value =
+        StackUtils.getArgument(frame, method.getBytecode(bytecodeIndex + 1),
+            method.getBytecode(bytecodeIndex + 2));
+
+    StackUtils.push(frame, value);
+
   }
 
-  private void doPushField(final int bytecodeIndex, final Frame frame, final SMethod method) {
-    // Handle the PUSH FIELD bytecode
+  private void doPushField(final int bytecodeIndex,
+      final SMethod method, final VirtualFrame frame) throws FrameSlotTypeException {
```

Comment:
What's the `FrameSlotTypeException` doing here? I'd think this needs to be handled else where.